### PR TITLE
WebSocketsClientHandler: Fix no await for websocket.close

### PR DESCRIPTION
**Pull request by roekatz**
@antoinetran If you can take a look as well I'll highly appreciate it

### CODE REVIEW

File: fastapi_websocket_rpc/websocket_rpc_client.py [hunk 1]
```diff
@@ -93,13 +106,17 @@ async def recv(self):
     async def close(self, code: int = 1000):
         if self._websocket is not None:
             # Case opened, we have something to close.
-            self._websocket.close(code)
+            await asyncio.get_event_loop().run_in_executor(
+                None, self._websocket.close, code
+            )
```
Comment (omer9564) [inline fastapi_websocket_rpc/websocket_rpc_client.py:111]: why does it matter if this code is blocking the event loop, it closes the websocket, why would you want anything else running in the same time ?

File: fastapi_websocket_rpc/websocket_rpc_client.py [hunk 2]
```diff
@@ -154,7 +173,8 @@ async def recv(self):
     async def close(self, code: int = 1000):
         if self._websocket is not None:
             # Case opened, we have something to close.
-            self._websocket.close(code)
+            await self._websocket.close(code)
```
Comment (omer9564) [inline fastapi_websocket_rpc/websocket_rpc_client.py:176]: can't websocket be None ?

File: fastapi_websocket_rpc/websocket_rpc_client.py [hunk 3]
```diff
@@ -93,13 +106,17 @@ async def recv(self):
     async def close(self, code: int = 1000):
         if self._websocket is not None:
             # Case opened, we have something to close.
-            self._websocket.close(code)
+            await asyncio.get_event_loop().run_in_executor(
+                None, self._websocket.close, code
```
Comment (omer9564) [inline fastapi_websocket_rpc/websocket_rpc_client.py:110]: if this is the same _websocket as in WebSocketsClientHandler, isn't at an async function already ?

File: fastapi_websocket_rpc/websocket_rpc_client.py [hunk 4]
```diff
@@ -165,7 +185,10 @@ def isNotForbbiden(value) -> bool:
     Returns:
         bool: Returns True as long as the given exception value is not InvalidStatusCode with 401 or 403
     """
-    return not (isinstance(value, InvalidStatusCode) and (value.status_code == 401 or value.status_code == 403))
+    return not (
+        isinstance(value, InvalidStatusCode)
+        and (value.status_code == 401 or value.status_code == 403)
```
Comment (omer9564) [inline fastapi_websocket_rpc/websocket_rpc_client.py:190]: would use value.status_code in (401,403)

File: fastapi_websocket_rpc/websocket_rpc_client.py (hunk 1 above)
Comment (roekatz) [reply fastapi_websocket_rpc/websocket_rpc_client.py:111]: I'm keeping with @antoinetran changes here - see other methods on this handler

File: fastapi_websocket_rpc/websocket_rpc_client.py (hunk 2 above)
Comment (roekatz) [reply fastapi_websocket_rpc/websocket_rpc_client.py:176]: There's `if self._websocket is not None:" just above...

File: fastapi_websocket_rpc/websocket_rpc_client.py (hunk 4 above)
Comment (roekatz) [reply fastapi_websocket_rpc/websocket_rpc_client.py:190]: That's just reformatting

**Review by omer9564 — APPROVED** [review]

**Comment by myroad94** [comment]
> @antoinetran If you can take a look as well I'll highly appreciate it

I'm in holiday, but looking at the code, it seems fine by me 🙂

**Pull request merged**